common/transport/amqp/src/amqp.ts

```diff
@@ -568,17 +866,29 @@ export class Amqp {
     }
   }
 
+  private _getErrorName(err: any): string {
+    if (err) {
+      if (err.condition) {
+        return '(amqp error) ' + err.condition;
+      } else if (err.hasOwnProperty('name')) {
+        return '(javascript error) ' + err.name;
+      } else {
+        return 'this is not an error type I understand';
```

Comment:
>'this is not an error type I understand' [](start = 15, length = 40)

I'm not sure how this helps anyone. who is "I"? why not?

Reply:
This code was ONLY used in debug statements.  I here refers to this function.  It's there because while I (in this case me) couldn't think of what would fall through to the else, it seemed more complete to put some kind of return for the else.

Changed test to 'unknown error type'

---
In reply to: [209133721](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209133721) [](ancestors = 209133721)